clientv3/grpc_options.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2017 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientv3
+
+import (
+	"math"
+
+	"google.golang.org/grpc"
+)
+
+var defaultCallOpts = []grpc.CallOption{
+	// Disable gRPC internal retrial logic
+	// TODO: enable when gRPC retry is stable (FailFast=false)
+	grpc.FailFast(true),
+
+	// client-side request send limit, gRPC default is math.MaxInt32
+	grpc.MaxCallSendMsgSize(math.MaxInt32),
```

Comment:
what is the sever side recv default limit? the send limit should be smaller than server side recv default limit.

Reply:
@xiang90 

Server-side default send `maxSendBytes` is `math.MaxInt32`.
Server-side default receive `embed.DefaultMaxRequestBytes` is 1.5MB.

I think for now, it is easier to set client-side parameters unlimited. Otherwise, this is easy to mis-configure. For example, user A configures 5MB server-side request limit, but forgot to configure client-side request receive limit. The write requests would go through, but receive request will fail.

Reply:
the goal of the default value is to prevent users to shoot in foot. I do not think we should expect users to send out a request that is > 4MB.

Reply:
@xiang90 Agree. But, https://github.com/kubernetes/kubernetes/issues/51099 is sending 26MB :0

Reply:
FWIW, The current rule of thumb for kubernetes is to limit all objects stored in etcd to under 1MB. So I'd expect sends to be in that range. Receives can be range reads, which can get pretty huge for things like `event-exporter` that literally dumps out the entire etcd events keyspace.